test/e2e/framework/dataplane.go

```diff
@@ -97,7 +113,7 @@ func verifyGlobalnetDatapathConnectivity(p tcp.ConnectivityTestParams) {
 	Expect(listenerPod.TerminationMessage).To(ContainSubstring(connectorPod.Config.Data))
 	Expect(stdOut).To(ContainSubstring(listenerPod.Config.Data))
 
-	if p.ToEndpointType == tcp.GlobalIP {
+	if p.ToEndpointType == tcp.GlobalServiceIP || p.ToEndpointType == tcp.GlobalPodIP {
```

Comment:
nit but L72-80 above assume `p.ToEndpointType` is one of these two so these checks shouldn't be necessary.